http/src/main.rs

```diff
@@ -66,35 +68,47 @@ fn main() {
                 std::process::exit(1);
             }
 
-            let bits = bits.get();
-
-            if bits < 1024 || bits > 16 * 1024 {
-                eprintln!("Error: --bits out of range [1024, 16384]: {}", bits);
-                eprintln!("This is a fake version of ipfs cli which does not support much");
-                std::process::exit(1);
-            }
-
-            if profile.len() != 1 || profile[0] != "test" {
-                eprintln!("Error: unsupported profile selection: {:?}", profile);
-                eprintln!("This is a fake version of ipfs cli which does not support much");
-                std::process::exit(1);
-            }
-
-            let result =
-                std::fs::create_dir_all(&home).and_then(|_| std::fs::File::create(&config_path));
+            let result = config::initialize(&home, bits, profile);
 
             match result {
                 Ok(_) => {
+                    let kp = std::fs::File::open(config_path)
+                        .map_err(config::LoadingError::ConfigurationFileOpening)
+                        .and_then(config::load)
+                        .unwrap();
+
                     // go-ipfs prints here (in addition to earlier "initializing ..."):
                     //
-                    // generating 2048-bit RSA keypair...done
-                    // peer identity: QmdNmxF88uyUzm8T7ps8LnCuZJzPnJvgUJxpKGqAMuxSQE
+                    // generating {}-bit RSA keypair...done
+
+                    println!("peer identity: {}", kp.public().into_peer_id());
                     std::process::exit(0);
                 }
-                Err(e) => {
+                Err(config::InitializationError::DirectoryCreationFailed(e)) => {
                     eprintln!("Error: failed to create repository path {:?}: {}", home, e);
                     std::process::exit(1);
                 }
+                Err(config::InitializationError::ConfigCreationFailed(_)) => {
+                    // this can be any number of errors like permission denied but these are the
+                    // strings from go-ipfs
+                    eprintln!("Error: ipfs configuration file already exists!");
+                    eprintln!("Reinitializing would override your keys.");
+                    std::process::exit(1);
```

Comment:
are the go ipfs error messages checked in the compliance test suite?

Reply:
I don't think so but I tried to include the ones I found. The might be presented in the conformance test assertion failures, at least if it happens during preparation.